internal/controllers/device_data_controller.go

```diff
@@ -30,7 +31,7 @@ type QueryDeviceErrorCodesReq struct {
 }
 
 type QueryDeviceErrorCodesResponse struct {
-	Message string `json:"message"`
+	Message []services.ErrorCodesResponse `json:"message"`
 }
 
 type GetUserDeviceErrorCodeQueriesResponse struct {
```

Comment:
Now, I'm worried about these response types. Shouldn't these change? I think this is the most serious concern I have.